bundles/org.eclipse.jface/src/org/eclipse/jface/wizard/WizardDialog.java

```diff
@@ -370,7 +370,7 @@ public WizardDialog setModal(boolean modal) {
 	 */
 	public boolean isModal() {
 		return (getShellStyle() & SWT.PRIMARY_MODAL) == SWT.PRIMARY_MODAL
-				|| (getShellStyle() & SWT.PRIMARY_MODAL) == SWT.APPLICATION_MODAL;
+				|| (getShellStyle() & SWT.APPLICATION_MODAL) == SWT.APPLICATION_MODAL;
```

Comment:
Because of this kind of hard to spot errors I recently suggested to use the style


```suggestion
				|| (getShellStyle() & SWT.APPLICATION_MODAL) != 0;
```

Reply:
I'm out of PC now, I believe all three variants have pros and cons, I will merge the original one, if you like, you can change to anything else later.